### PR TITLE
V14 QA added Scripts acceptance tests

### PR DESCRIPTION
**Pull request by andr317c**
Updated API test for the scripts to work with the updated testHelpers
Added Script UI acceptance test
### To test
Follow the README.md located in tests/Umbraco.Tests.AcceptanceTest for running the test
Test the tests in tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Settings/Scripts.spec.ts

### CODE REVIEW

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Settings/Scripts.spec.ts [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+  const scriptPath = scriptName + '.js';
+  const scriptFolderName = 'TestScriptFolder';
+
+  test('can create a empty script', async ({page, umbracoApi, umbracoUi}) => {
+    // Arrange
+    await umbracoApi.script.ensureNameNotExists(scriptPath);
+
+    // Act
+    await page.goto(umbracoApi.baseUrl + '/umbraco');
+    await umbracoUi.goToSection(ConstantHelper.sections.settings);
```
Comment (nhudinh0309) [inline tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Settings/Scripts.spec.ts:16]: it always run these piece of code before all tests, so should be put in beforeEach()
     await page.goto(umbracoApi.baseUrl + '/umbraco');
    await umbracoUi.goToSection(ConstantHelper.sections.settings);

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Settings/Scripts.spec.ts [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+    await page.locator('umb-tree-item', {hasText: 'Scripts'}).getByLabel('Open actions menu').click({force: true});
+    await page.getByLabel('New empty script').click();
+    // TODO: Change the label to script name when the label is updated
+    await page.getByLabel('template name').fill(scriptName);
+    // TODO: Remove this timeout when frontend validation is implemented
+    await page.waitForTimeout(1000);
```
Comment (nhudinh0309) [inline tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Settings/Scripts.spec.ts:22]: You can use await page.waitForLoadState('networkidle') to wait for the network to idle till it loads

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Settings/Scripts.spec.ts [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+    await page.locator('umb-tree-item').locator('[label="' + scriptFolderName + '"] >> [label="Open actions menu"]').click();
+    await page.getByLabel('New empty script').click();
+    // TODO: Change the label to script name when the label is updated
+    await page.getByLabel('template name').fill(scriptName);
+    await page.locator('textarea.inputarea').clear();
+    await page.locator('textarea.inputarea').fill(scriptContent);
+    await page.getByLabel('Save').click();
+    // TODO: Remove this timeout when frontend validation is implemented
+    await page.waitForTimeout(1000);
+    await page.getByLabel('Save').click({force: true});
```
Comment (nhudinh0309) [inline tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Settings/Scripts.spec.ts:145]: why it need to click Save button twice?

**Review by nhudinh0309 — COMMENTED** [review]
I added few comments

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Settings/Scripts.spec.ts (hunk 2 above)
Comment (andr317c) [reply tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Settings/Scripts.spec.ts:24]: I'm not totally sure if it is a good way to go 🙁 Playwright themselves discourage the use of 'networkidle'

Comment (nhudinh0309) [reply tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Settings/Scripts.spec.ts:24]: you're right. so just ignore this comment.

**Comment by nhudinh0309** [comment]
Looks great 💪 Merging

**Pull request merged**